docs/public/installing-and-onfiguring-minikube-environment.adoc

```diff
@@ -0,0 +1,131 @@
+= Installing and configuring minikube environment
+
+[NOTE]
+----
+The guide only works with minikube configured with kubernetes 1.19.x or lower. `odo link` cannot link services successfully in a Kubernetes 1.20.x or newer environment.
+----
+
+It is recommended that users of this guide obtain a suitable system for running minikube with kubernetes. In practice this should be a 4 core system minimum. Before proceeding to the sample application, please follow the instructions for establishing a minikube environment:
+
+.Prerequisites
+* You have installed docker. See link:https://docs.docker.com/engine/install/[the Docker documentation].
+
+* You have installed minikube. See link:https://minikube.sigs.k8s.io/docs/start/[the minikube installation instructions] for your operating system.
+
+=== Starting minikube
+
+* If you try to run minikube as a root user, minikube throws an error that docker should not be run as root and aborts the startup. To proceed, start minikube in a manner that overrides this protection mechanism:
++
+[source,sh]
+----
+$ minikube start --force --driver=docker --kubernetes-version=v1.19.8
+----
+
+* If you are a non-root user, start minikube as usual:
++
+[source,sh]
+----
+$ minikube start --kubernetes-version=v1.19.8
+----
+
+=== Configuring minikube
+
+==== Enabling ingress addon 
+
+The application requires an ingress addon to allow the routes to be created easily. It enables `odo url` commands. 
+
+.. onfigure minikube for ingress by adding ingress as a minikube add-on:
++
+[source,sh]
+----
+$ minikube addons enable ingress
+----
+
+==== Adding a pull secret to ingress accounts
+
+You may face the DockerHub pull rate limit if you do not have a pull secret for your personal free DockerHub account. During ingress initialization two of the job pods used by ingress may fail to initialize due to pull rate limits. If this happens, and ingress fails to enable, you add a secret for the pulls for the following service accounts:
+
+* ingress-nginx-admission
+* ingress-nginx
+
+to add a pull secret for these service accounts: 
+
+. Switch to the kube-system context:
++
+[source,sh]
+----
+$ kubectl config set-context --current --namespace=kube-system
+----
+
+. Create a pull secret:
++
+[source,sh]
+----
+$ kubectl create secret docker-registry regcred --docker-server=<your-registry-server> --docker-username=<your-name> --docker-password=<your-pword> --docker-email=<your-email>
+----
++
+where:
++
+* <your-registry-server> is the DockerHub Registry FQDN. (https://index.docker.io/v1/)
+* your-name> is your Docker username.
+* <your-pword> is your Docker password.
+* <your-email> is your Docker email.
```

Comment:
Nitpick - aren't these "Docker Hub" configurations? Something like "email used for Docker Hub account" would make more sense, maybe?